edweiss-app/__test__/app/(app)/lectures/remotecontrol/index.test.tsx

```diff
@@ -0,0 +1,144 @@
+        (usePrefetchedDynamicDoc as jest.Mock).mockReturnValue([undefined]);
+        const { getByTestId } = render(<RemoteControlScreen />);
+        expect(getByTestId('activity-indicator')).toBeTruthy();
+    });
+
+    it('renders the main screen when lectureDoc is available', () => {
+        const { getByText } = render(<RemoteControlScreen />);
+        expect(getByText("Lecture's Slides")).toBeTruthy();
+    });
+
+    it('sets screen orientation to portrait when lectureDoc is loaded', async () => {
+        render(<RemoteControlScreen />);
+        await waitFor(() => {
+            expect(ScreenOrientation.lockAsync).toHaveBeenCalledWith(ScreenOrientation.OrientationLock.PORTRAIT_UP);
+        });
+    });
+
+    it('starts and stops recording when mic button is pressed', async () => {
+        const { getByTestId } = render(<RemoteControlScreen />);
+        const micButton = getByTestId('mic-button');
+
+        // Start recording
+        fireEvent.press(micButton);
+        await waitFor(() => expect(Voice.start).toHaveBeenCalledWith('en-US'));
+        expect(Voice.start).toHaveBeenCalledTimes(1);
+
+        // Stop recording
+        fireEvent.press(micButton);
+        await waitFor(() => expect(Voice.stop).toHaveBeenCalled());
+        expect(Voice.stop).toHaveBeenCalledTimes(1);
+    });
+
+    it('navigates to the next slide and starts/stops recording appropriately', async () => {
+        const { getByTestId } = render(<RemoteControlScreen />);
+        const nextButton = getByTestId('next-button');
+
+        fireEvent.press(nextButton);
+        expect(callFunction).toHaveBeenCalledWith(expect.any(String), { page: 2 });
+        expect(Vibration.vibrate).toHaveBeenCalledWith(100);
+    });
+
+    it('navigates to the previous slide and starts/stops recording appropriately', async () => {
+        const { getByTestId } = render(<RemoteControlScreen />);
+        const prevButton = getByTestId('prev-button');
+
+        fireEvent.press(prevButton);
+        expect(callFunction).toHaveBeenCalledWith(expect.any(String), { page: 1 });
+        expect(Vibration.vibrate).toHaveBeenCalledWith(100);
+    });
+
+    it('handles audio transcription and updates the slide transcription', async () => {
+        const mockSpeechResults = { value: ['Test transcription'] };
+        const { getByTestId } = render(<RemoteControlScreen />);
+
+        // Trigger onSpeechResults with mock data
+        await waitFor(() => Voice.onSpeechResults(mockSpeechResults));
+
+        expect(callFunction).toHaveBeenCalledWith('addAudioTranscript', {
+            courseId: 'testCourse',
+            lectureId: 'testLectureId',
+            pageNumber: 1,
+            transcription: 'Test transcription'
+        });
+    });
+
+    // it('handles errors in transcription', async () => {
+    //     const mockError = { error: 'An error occurred' };
+    //     const consoleSpy = jest.spyOn(console, 'log');
```

Comment:
You can delete this before merge, it could be a good idee to increase your personal score for the course

Reply:
Thank's for pointing out. I do it right away.